src/main/scala/scala/collection/decorators/IteratorDecorator.scala

```diff
@@ -72,4 +72,47 @@ class IteratorDecorator[A](val `this`: Iterator[A]) extends AnyVal {
     loop(immutable.List.empty)
   }
 
+  /**
+    * Constructs an iterator where consecutive elements are accumulated as
+    * long as the output of f for each element doesn't change.
+    * <pre>
+    * Vector(1,2,2,3,3,3,2,2)
+    *     .iterator
+    *     .groupUntilChanged(identity)
+    *     .toList
+    * </pre>
+    * produces
+    * <pre>
+    * List(Seq(1),
+    *      Seq(2,2),
+    *      Seq(3,3,3),
+    *      Seq(2,2))
+    * </pre>
+    * @param f the function to compute a key for an element
+    * @tparam K the type of the computed key
+    * @return an iterator of sequences of the consecutive elements with the
+    *         same key in the original iterator
+    */
+  def groupUntilChanged[K](f: A => K): Iterator[Iterable[A]] =
+    new AbstractIterator[Seq[A]] {
+      private var hd: Option[A] = readHead()
```

Comment:
This variable could be named `maybeHead` or something that shows a better intent

Reply:
Variables with the same kind of usage are named like this in the std lib (search var hd in https://github.com/scala/scala/blob/2.13.x/src/library/scala/collection/Iterator.scala) so I chose to remain consistent. Arguably I should have stayed consistent all the way through and used a nullable variable and a boolean but I didn't feel very confortable with that)